Core/GDCore/Project/Variable.cpp

```diff
@@ -10,35 +10,120 @@
 
 #include "GDCore/Serialization/SerializerElement.h"
 #include "GDCore/String.h"
-#include "GDCore/TinyXml/tinyxml.h"
 
 using namespace std;
 
 namespace gd {
 
-/**
- * Get value as a double
- */
+gd::Variable Variable::badVariable;
+
+gd::String Variable::TypeAsString(Type t) {
+  switch (t) {
+    case Type::String:
+      return "string";
+    case Type::Number:
+      return "number";
+    case Type::Boolean:
+      return "boolean";
+    case Type::Structure:
+      return "structure";
+    case Type::Array:
+      return "array";
+    default:
+      return "error-type";
+  }
+};
+
+Variable::Type Variable::StringAsType(const gd::String& str) {
+  if (str == "string")
+    return Type::String;
+  else if (str == "number")
+    return Type::Number;
+  else if (str == "boolean")
+    return Type::Boolean;
+  else if (str == "structure")
+    return Type::Structure;
+  else if (str == "array")
+    return Type::Array;
+
+  // Default to number
+  return Type::Number;
+}
+
+bool Variable::IsPrimitive(const Type type) {
+  return type == Type::String || type == Type::Number || type == Type::Boolean;
+}
+
+void Variable::CastTo(const Type newType) {
+  if (newType == Type::Number)
+    SetValue(GetValue());
+  else if (newType == Type::String)
+    SetString(GetString());
+  else if (newType == Type::Boolean)
+    SetBool(GetBool());
+  else if (newType == Type::Structure) {
+    children.clear();
+
+    // Conversion is only possible for non prmitive types
+    if (type == Type::Array)
+      for (auto i = childrenArray.begin(); i != childrenArray.end(); ++i)
+        children.insert(
+            std::make_pair(gd::String::From(i - childrenArray.begin()), (*i)));
+
+    type = Type::Structure;
+  } else if (newType == Type::Array) {

```

Comment:
Do you think that because we've now casted to a structure, we can clear childrenArray?